Bumping version

Clarkcj12 opened the pull request:
Attempting to update plugin

Clarkcj12 committed:
Merge pull request #1 from WirelessRedstoneGroup/master

Update

Clarkcj12 committed:
Merge branch 'WirelessRedstoneGroup:master' into master

dependabot[bot] committed:
Bump commons-io from 2.5 to 2.7 in /core

Bumps commons-io from 2.5 to 2.7.

---
updated-dependencies:
- dependency-name: commons-io:commons-io
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Clarkcj12 committed:
Merge pull request #2 from Clarkcj12/dependabot/maven/core/commons-io-commons-io-2.7

Bump commons-io from 2.5 to 2.7 in /core

Clarkcj12 committed:
Create dependabot.yml

dependabot[bot] committed:
Bump maven-compiler-plugin from 3.8.1 to 3.10.1

Bumps [maven-compiler-plugin](https://github.com/apache/maven-compiler-plugin) from 3.8.1 to 3.10.1.
- [Release notes](https://github.com/apache/maven-compiler-plugin/releases)
- [Commits](https://github.com/apache/maven-compiler-plugin/compare/maven-compiler-plugin-3.8.1...maven-compiler-plugin-3.10.1)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-compiler-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump slf4j-simple from 1.7.25 to 1.7.36

Bumps [slf4j-simple](https://github.com/qos-ch/slf4j) from 1.7.25 to 1.7.36.
- [Release notes](https://github.com/qos-ch/slf4j/releases)
- [Commits](https://github.com/qos-ch/slf4j/compare/v_1.7.25...v_1.7.36)

---
updated-dependencies:
- dependency-name: org.slf4j:slf4j-simple
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump worldedit-core from 6.1 to 7.2.8

Bumps worldedit-core from 6.1 to 7.2.8.

---
updated-dependencies:
- dependency-name: com.sk89q.worldedit:worldedit-core
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump worldedit-bukkit from 6.1 to 7.2.8

Bumps worldedit-bukkit from 6.1 to 7.2.8.

---
updated-dependencies:
- dependency-name: com.sk89q.worldedit:worldedit-bukkit
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump annotations from 13.0 to 23.0.0

Bumps [annotations](https://github.com/JetBrains/java-annotations) from 13.0 to 23.0.0.
- [Release notes](https://github.com/JetBrains/java-annotations/releases)
- [Changelog](https://github.com/JetBrains/java-annotations/blob/master/CHANGELOG.md)
- [Commits](https://github.com/JetBrains/java-annotations/commits/23.0.0)

---
updated-dependencies:
- dependency-name: org.jetbrains:annotations
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Clarkcj12 committed:
Merge pull request #3 from Clarkcj12/dependabot/maven/org.apache.maven.plugins-maven-compiler-plugin-3.10.1

Bump maven-compiler-plugin from 3.8.1 to 3.10.1

Clarkcj12 committed:
Merge pull request #4 from Clarkcj12/dependabot/maven/org.slf4j-slf4j-simple-1.7.36

Bump slf4j-simple from 1.7.25 to 1.7.36

Clarkcj12 committed:
Merge pull request #5 from Clarkcj12/dependabot/maven/com.sk89q.worldedit-worldedit-core-7.2.8

Bump worldedit-core from 6.1 to 7.2.8

Clarkcj12 committed:
Merge pull request #6 from Clarkcj12/dependabot/maven/com.sk89q.worldedit-worldedit-bukkit-7.2.8

Bump worldedit-bukkit from 6.1 to 7.2.8

Clarkcj12 committed:
Merge pull request #7 from Clarkcj12/dependabot/maven/org.jetbrains-annotations-23.0.0

Bump annotations from 13.0 to 23.0.0

dependabot[bot] committed:
Bump commons-io from 2.7 to 2.11.0

Bumps commons-io from 2.7 to 2.11.0.

---
updated-dependencies:
- dependency-name: commons-io:commons-io
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump gson from 2.8.5 to 2.9.0

Bumps [gson](https://github.com/google/gson) from 2.8.5 to 2.9.0.
- [Release notes](https://github.com/google/gson/releases)
- [Changelog](https://github.com/google/gson/blob/master/CHANGELOG.md)
- [Commits](https://github.com/google/gson/compare/gson-parent-2.8.5...gson-parent-2.9.0)

---
updated-dependencies:
- dependency-name: com.google.code.gson:gson
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump maven-shade-plugin from 3.2.1 to 3.3.0

Bumps [maven-shade-plugin](https://github.com/apache/maven-shade-plugin) from 3.2.1 to 3.3.0.
- [Release notes](https://github.com/apache/maven-shade-plugin/releases)
- [Commits](https://github.com/apache/maven-shade-plugin/compare/maven-shade-plugin-3.2.1...maven-shade-plugin-3.3.0)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-shade-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Clarkcj12 committed:
Merge pull request #10 from Clarkcj12/dependabot/maven/org.apache.maven.plugins-maven-shade-plugin-3.3.0

Bump maven-shade-plugin from 3.2.1 to 3.3.0

Clarkcj12 committed:
Merge pull request #9 from Clarkcj12/dependabot/maven/com.google.code.gson-gson-2.9.0

Bump gson from 2.8.5 to 2.9.0

Clarkcj12 committed:
Merge pull request #8 from Clarkcj12/dependabot/maven/commons-io-commons-io-2.11.0

Bump commons-io from 2.7 to 2.11.0

dependabot[bot] committed:
Bump gson from 2.9.0 to 2.9.1

Bumps [gson](https://github.com/google/gson) from 2.9.0 to 2.9.1.
- [Release notes](https://github.com/google/gson/releases)
- [Changelog](https://github.com/google/gson/blob/master/CHANGELOG.md)
- [Commits](https://github.com/google/gson/compare/gson-parent-2.9.0...gson-parent-2.9.1)

---
updated-dependencies:
- dependency-name: com.google.code.gson:gson
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump slf4j-simple from 1.7.36 to 2.0.0

Bumps [slf4j-simple](https://github.com/qos-ch/slf4j) from 1.7.36 to 2.0.0.
- [Release notes](https://github.com/qos-ch/slf4j/releases)
- [Commits](https://github.com/qos-ch/slf4j/compare/v_1.7.36...v_2.0.0)

---
updated-dependencies:
- dependency-name: org.slf4j:slf4j-simple
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Clarkcj12 committed:
Merge pull request #11 from Clarkcj12/dependabot/maven/com.google.code.gson-gson-2.9.1

Bump gson from 2.9.0 to 2.9.1

Clarkcj12 committed:
Merge pull request #12 from Clarkcj12/dependabot/maven/org.slf4j-slf4j-simple-2.0.0

Bump slf4j-simple from 1.7.36 to 2.0.0

dependabot[bot] committed:
Bump maven-shade-plugin from 3.3.0 to 3.4.0

Bumps [maven-shade-plugin](https://github.com/apache/maven-shade-plugin) from 3.3.0 to 3.4.0.
- [Release notes](https://github.com/apache/maven-shade-plugin/releases)
- [Commits](https://github.com/apache/maven-shade-plugin/compare/maven-shade-plugin-3.3.0...maven-shade-plugin-3.4.0)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-shade-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump slf4j-simple from 2.0.0 to 2.0.1

Bumps [slf4j-simple](https://github.com/qos-ch/slf4j) from 2.0.0 to 2.0.1.
- [Release notes](https://github.com/qos-ch/slf4j/releases)
- [Commits](https://github.com/qos-ch/slf4j/compare/v_2.0.0...v_2.0.1)

---
updated-dependencies:
- dependency-name: org.slf4j:slf4j-simple
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Clarkcj12 committed:
Merge pull request #13 from Clarkcj12/dependabot/maven/org.apache.maven.plugins-maven-shade-plugin-3.4.0

Bump maven-shade-plugin from 3.3.0 to 3.4.0

Clarkcj12 committed:
Merge pull request #14 from Clarkcj12/dependabot/maven/org.slf4j-slf4j-simple-2.0.1

Bump slf4j-simple from 2.0.0 to 2.0.1

axelpey commented:
@Clarkcj12 how does it work with your updates ? Is the problem related to simply updating dependencies ?

dependabot[bot] committed:
Bump slf4j-simple from 2.0.1 to 2.0.3

Bumps [slf4j-simple](https://github.com/qos-ch/slf4j) from 2.0.1 to 2.0.3.
- [Release notes](https://github.com/qos-ch/slf4j/releases)
- [Commits](https://github.com/qos-ch/slf4j/compare/v_2.0.1...v_2.0.3)

---
updated-dependencies:
- dependency-name: org.slf4j:slf4j-simple
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Clarkcj12 committed:
Merge pull request #16 from Clarkcj12/dependabot/maven/org.slf4j-slf4j-simple-2.0.3

Bump slf4j-simple from 2.0.1 to 2.0.3

dependabot[bot] committed:
Bump maven-shade-plugin from 3.4.0 to 3.4.1

Bumps [maven-shade-plugin](https://github.com/apache/maven-shade-plugin) from 3.4.0 to 3.4.1.
- [Release notes](https://github.com/apache/maven-shade-plugin/releases)
- [Commits](https://github.com/apache/maven-shade-plugin/compare/maven-shade-plugin-3.4.0...maven-shade-plugin-3.4.1)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-shade-plugin
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump slf4j-simple from 2.0.3 to 2.0.5

Bumps [slf4j-simple](https://github.com/qos-ch/slf4j) from 2.0.3 to 2.0.5.
- [Release notes](https://github.com/qos-ch/slf4j/releases)
- [Commits](https://github.com/qos-ch/slf4j/compare/v_2.0.3...v_2.0.5)

---
updated-dependencies:
- dependency-name: org.slf4j:slf4j-simple
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Clarkcj12 committed:
Merge pull request #17 from Clarkcj12/dependabot/maven/org.apache.maven.plugins-maven-shade-plugin-3.4.1

Bump maven-shade-plugin from 3.4.0 to 3.4.1

Clarkcj12 committed:
Merge pull request #19 from Clarkcj12/dependabot/maven/org.slf4j-slf4j-simple-2.0.5

Bump slf4j-simple from 2.0.3 to 2.0.5

dependabot[bot] committed:
Bump slf4j-simple from 2.0.5 to 2.0.6

Bumps [slf4j-simple](https://github.com/qos-ch/slf4j) from 2.0.5 to 2.0.6.
- [Release notes](https://github.com/qos-ch/slf4j/releases)
- [Commits](https://github.com/qos-ch/slf4j/compare/v_2.0.5...v_2.0.6)

---
updated-dependencies:
- dependency-name: org.slf4j:slf4j-simple
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump annotations from 23.0.0 to 24.0.0

Bumps [annotations](https://github.com/JetBrains/java-annotations) from 23.0.0 to 24.0.0.
- [Release notes](https://github.com/JetBrains/java-annotations/releases)
- [Changelog](https://github.com/JetBrains/java-annotations/blob/master/CHANGELOG.md)
- [Commits](https://github.com/JetBrains/java-annotations/compare/23.0.0...24.0.0)

---
updated-dependencies:
- dependency-name: org.jetbrains:annotations
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Clarkcj12 committed:
Merge pull request #21 from Clarkcj12/dependabot/maven/org.slf4j-slf4j-simple-2.0.6

Bump slf4j-simple from 2.0.5 to 2.0.6

Clarkcj12 committed:
Merge pull request #22 from Clarkcj12/dependabot/maven/org.jetbrains-annotations-24.0.0

Bump annotations from 23.0.0 to 24.0.0

Clarkcj12 committed:
Update Java version, Minecraft version, and dependency

Upgraded the Java compiler target and source to version 21 and updated the Minecraft version to 1.20.6. Additionally, bumped the JetBrains annotations dependency to version 26.0.2. These changes ensure compatibility with newer features and dependencies.

Clarkcj12 committed:
Upgrade dependencies and update Maven configurations

Updated Maven compiler plugin to version 3.14.0 and upgraded Java source/target compatibility. Bumped Spigot, WorldEdit, and additional dependencies to newer versions. Fixed repository URLs and corrected a typo in comments for better clarity.

Clarkcj12 committed:
Update Maven compiler plugin version and Java target/source

Upgraded the maven-compiler-plugin to version 3.14.0 and updated the Java source and target compatibility from 1.8 to 22. This ensures compatibility with newer Java features and aligns the build configuration with modern standards.

Clarkcj12 committed:
Update dependencies and plugin versions in pom files

Updated the Spigot API version to 1.13.2-R0.1-SNAPSHOT for spigot_1_13. Adjusted Maven compiler plugin to version 3.14.0 with Java 22 support for spigot_1_8. Upgraded Maven shade plugin to version 3.6.0 in the dist module.

Clarkcj12 committed:
Update API version to 1.20.6 in plugin.yml

Upgraded the `api-version` from 1.13 to 1.20.6 to ensure compatibility with newer server versions. This change helps maintain the plugin's functionality on modern platforms.

Clarkcj12 committed:
Switch to Paper API and update dependencies

Replaced Spigot API with Paper API across the project for improved performance and features. Updated multiple dependencies (Gson, Commons IO, SLF4J) to newer versions. Transitioned repositories from Spigot to PaperMC's Maven repository. Updated project semantic version to 0.10.2.

Clarkcj12 committed:
Refactor onDisable to improve cleanup and logging.

Added safeguards to close storage, unregister hooks, shut down sentry, and reset static references to prevent memory leaks. Enhanced logging to confirm proper plugin disablement.

Clarkcj12 committed:
Handle missing or unreadable plugin.yml gracefully

Ensure the plugin disables itself if plugin.yml is missing or throws an IOException during loading. Added logging for better error debugging and prevention of undefined behavior.

Clarkcj12 committed:
Improve incompatibility logging for better debugging

Added detailed server and plugin version information in the logging when the plugin is incompatible with the server. This helps users identify version mismatches more easily and directs them to the plugin's page for supported versions.

Clarkcj12 committed:
Refactor main plugin structure and improve initialization

Simplified imports by using wildcard and improved organization of methods for better readability. Added robust initialization and shutdown procedures for storage, commands, metrics, and third-party integrations like WorldEdit and Sentry. Enhanced logging and error handling for compatibility checks, update checks, and event management.

Clarkcj12 committed:
Ensure null safety in command registration logic

Added `Objects.requireNonNull` to enforce null checks when registering command executors and tab completers. This prevents potential null pointer exceptions during plugin initialization.

Clarkcj12 committed:
Refactor WRLogger to use Optional for console field.

Replaced direct usage of ConsoleCommandSender with Optional to handle potential null values more safely. This improves code reliability and aligns with modern Java best practices.

Clarkcj12 committed:
Refactor WRLogger for safer console handling with Optional

Replaced direct ConsoleCommandSender reference with Optional for null safety. Simplified the prefix initialization logic for better readability and maintainability.

Clarkcj12 committed:
Refactor WRLogger to use Kyori Adventure API.

Replaces Bukkit's console messaging with Kyori Adventure API for improved text formatting and message handling. Simplifies logic with a unified `log` method and streamlined color handling, while maintaining backward compatibility via plain text fallback. Adds Adventure API as a project dependency.

Clarkcj12 committed:
Refactor logging and code structure for clarity

Replaced legacy logger with WRLogger using Adventure API for improved message formatting and clarity. Simplified methods, reduced redundancy, and improved modularity across initialization, event registration, and command handling. Adjusted naming conventions to enhance readability.

Clarkcj12 committed:
Refactor to integrate Adventure API for logging and messaging

Replaced legacy console-based logging with Adventure API for improved flexibility and centralized handling. Added support for `BukkitAudiences`, updated event registration, and streamlined resource cleanup during plugin disable. Enhanced command registration with missing-command checks and improved error handling.

Clarkcj12 committed:
Enhance plugin initialization and lifecycle management

Replaced deprecated plugin disabling via `getPluginLoader()` with a clean custom method. Added Adventure platform dependency to support better logging. Improved code comments for clarity and ensured early exits for error cases.

Clarkcj12 committed:
Refactor Utils method calls to support additional parameters

Updated `Utils.getSignType` to `Utils.getType` across multiple classes, adding support for a second parameter (line 2) in method calls. This enhances functionality and ensures consistency in sign type determination logic throughout the codebase.

Clarkcj12 committed:
Refactor `Utils` for improved code clarity and efficiency.

Replaced redundant logic with streamlined methods, leveraging Java 8+ features like `List.of`, `Map.of`, and `switch` expressions. Optimized version parsing, message handling, and direction conversions to enhance maintainability and readability.

Clarkcj12 committed:
Refactor UpdateChecker to use modern HttpClient API

Replaced outdated HttpURLConnection with Java's HttpClient for asynchronous update checks, improving readability and code maintainability. Enhanced thread safety in singleton initialization and data access. Adjusted response parsing and error handling for robustness and clarity.

Clarkcj12 committed:
Refactor logging and add `getStrings` method.

Updated `WRLogger` initialization to cast `audience.console()` to `ConsoleCommandSender` for improved type safety. Introduced `getStrings` method to enhance string management accessibility.

Clarkcj12 committed:
Refactor utility methods and improve documentation.

Refactored `Utils` class methods for improved clarity, better parameter handling, and consistency. Enhanced Javadoc comments across various methods with additional details, formatting, and explanations. Marked the `getRawData` method as deprecated for future removal.

Clarkcj12 committed:
Simplify imports in UpdateChecker.java

Removed an unused InputStreamReader import to clean up the code. This enhances maintainability and eliminates unnecessary dependencies.

Clarkcj12 committed:
Add StorageConfiguration initialization and accessor method

Introduce proper initialization for StorageConfiguration via StorageManager and ensure storage setup is verified during plugin startup. Added `getStorage()` method to provide safe access to the storage configuration, with a validation check for uninitialized states.

Clarkcj12 committed:
Fix logical flow in hasAccessToChannel method

Refactored the method to ensure proper handling of player checks. Now always returns true for non-player senders like console or command blocks, preserving intended behavior.

Clarkcj12 committed:
Add stub for sendFeedback method in Utils

Introduce the sendFeedback method with a placeholder implementation. This prepares the code for future enhancements requiring feedback handling functionality.

Clarkcj12 committed:
Refactor AdminAddOwner command with enhanced validations.

Improved argument validation, added specific error feedback, and encapsulated owner addition logic into a new method for better error handling. These changes enhance maintainability and provide clearer notifications for command users.

Clarkcj12 committed:
Add Javadoc comments for AdminAddOwner command method

Added detailed Javadoc comments to the onCommand method in AdminAddOwner class. This includes explanation of the method's purpose and parameters, improving code readability and maintainability.

dependabot[bot] committed:
Bump io.sentry:sentry from 1.7.23 to 8.3.0

Bumps [io.sentry:sentry](https://github.com/getsentry/sentry-java) from 1.7.23 to 8.3.0.
- [Release notes](https://github.com/getsentry/sentry-java/releases)
- [Changelog](https://github.com/getsentry/sentry-java/blob/main/CHANGELOG.md)
- [Commits](https://github.com/getsentry/sentry-java/compare/v1.7.23...8.3.0)

---
updated-dependencies:
- dependency-name: io.sentry:sentry
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Clarkcj12 committed:
Merge pull request #34 from Clarkcj12/dependabot/maven/io.sentry-sentry-8.3.0

Bump io.sentry:sentry from 1.7.23 to 8.3.0

RufiS commented:
.
